sound/soc/sof/sof-audio.h

```diff
@@ -38,6 +38,8 @@
 enum sof_widget_op {
 	SOF_WIDGET_FREE,
 	SOF_WIDGET_SETUP,
+	SOF_WIDGET_PREPARE,
+	SOF_WIDGET_UNPREPARE,
```

Comment:
you really ought to describe what those states mean.

It seems from reading the code that PREPARE is done before SETUP, but I have no idea how UNPREPARE/FREE are related.

Reply:
I've updated the commit message and add some comments. Hopefully they help.

Reply:
I would change the order to describe the logical use of these enums

enum sof_widget_op {
SOF_WIDGET_PREPARE,
 SOF_WIDGET_SETUP,
 SOF_WIDGET_FREE,
 SOF_WIDGET_UNPREPARE,
}

Reply:
And what are these exactly?
Are they
a) `state`s
b) `operation`s (`op`s)
c) `action`s
d) something else?

Reply:
operations to perform while walking the widgets, ie set up the widgets while walking, free them or prepare them